### ISSUE TITLE
ParseNode Failure while processing string literal

### ISSUE THREAD

**Issue by dadams39** [microsoft/kiota-serialization-json-go#30]
Present on **github.com/microsoft/kiota-serialization-json-go**  `v0.5.1` & `v0.5.2`
Related to: https://github.com/microsoft/kiota-serialization-json-go/pull/26

call: `js.NewJsonParseNodeFactory().GetRootParseNode("application/json", bytes)`
where
### bytes
```
{"id":"AAMkADhlNjU5MzMzLTUyYzItNDJmYi04ZTgzLTkzYTE1ZDg0OTRjOQBGAAAAAAC9muvJZ4uvRYeQkyPxt40qBwCDUn-uvuceTb3Zb9F345Q1AAAAAAEMAACDUn-uvuceTb3Zb9F345Q1AAAuZPpRAAA=","@odata.context":"https://graph.microsoft.com/v1.0/$metadata#users('67fc108d-cc22-4a54-96af-7b91c5d34cf2')/messages/$entity","@odata.etag":"W/\"CQAAABYAAACDUn/uvuceTb3Zb9F345Q1AAAuT0ok\"","categories":[],"changeKey":"CQAAABYAAACDUn/uvuceTb3Zb9F345Q1AAAuT0ok","createdDateTime":"2022-06-15T05:27:05Z","lastModifiedDateTime":"2022-06-15T05:27:07Z","bccRecipients":[],"body":{"content":"<html lang=\"en\" style=\"min-height:100%; background:#ffffff\"><head>
<meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\"><meta name=\"viewport\" content=\"width=device-width\"><meta name=\"eventId\" content=\"aad-identity-protection-weekly-digest-report-v2\"><meta name=\"messageId\" content=\"d4db577e-fe10-4bea-8e6d-164bb1ebb039\"><meta name=\"cloud\" content=\"AZ\"><style id=\"mediaqueries\">
<!--
@media only screen and (max-width: 640px) {
.wrap-dangler
	{padding-right:10%!important}
.wrap-dangler.small-text-center .wrap-dangler.text-center
```
Receiving error: `invalid character '\r' in string literal `

**Comment by dadams39** [comment]
## Error Tracking
 **json_parse_node.go**
During `func loadJsonTree(decoder *json.Decoder) (*JsonParseNode, error)`
**line 54:**  
> childNode, err := loadJsonTree(decoder)

**context:** `keyStr:` "body

`encoding/json.SyntaxError {msg: "invalid character '\\r' in string literal", Offset: 592}`

**Comment by baywet** [comment]
Hi @dadams39,
Thank you for reporting this.
Would you be willing to submit a PR to address this and https://github.com/microsoftgraph/msgraph-sdk-go/issues/203 (\ needs to be replaced first before anything else) ?

**Comment by dadams39** [comment]
From what I can tell, the Serialization Writer creates these types of messages incorrectly. I'm unable to verify the original data store input. The ParseNode failing to write what is an invalid `JSON` object is the correct outcome. Going to need to find a test case that I can reference with the Graph Explorer to be able to find a good example to fix the parsing error.

**Comment by baywet** [comment]
hopefully the lengthy explanation I've added in #28 clarifies the situation.
I do believe we should still escape \t in the serialization writer.

**Comment by dadams39** [comment]
Returning to this as the following has been closed: https://github.com/microsoftgraph/msgraph-sdk-go/issues/203. The error remains unchanged.

**Comment by baywet** [comment]
ah, I can see you left a thumbs up. This doesn't send out a notification to catch my attention back. Thanks for confirming.
Is this something you'd be willing to submit a pull request for?

https://github.com/microsoft/kiota-serialization-json-go/blob/996ac74ff5a5eeb3554e3b1940030d8cadf4880c/json_serialization_writer.go#L37

https://github.com/microsoft/kiota-serialization-json-go/blob/996ac74ff5a5eeb3554e3b1940030d8cadf4880c/json_serialization_writer_test.go#L189

**Comment by dadams39** [comment]
The tab is showing up as the issue, but the true failure may be the `\r` carriage return. I am working to see the target file if I can create a valid JSON making these changes. I will be submitting the PR soon.

**Comment by dadams39** [comment]
Can be closed after PR #34  is merged
